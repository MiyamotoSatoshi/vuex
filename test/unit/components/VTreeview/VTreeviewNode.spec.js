import Vue from 'vue'
import { test } from '@/test'
import VTreeviewNode from '@/components/VTreeview/VTreeviewNode'

const vm = new Vue()
const defaultSlot = () => vm.$createElement('div', 'foobar')

const Mock = {
  name: 'test',

  render: h => h(VTreeviewNode, {
    scopedSlots: {
      prepend: defaultSlot,
      append: defaultSlot
    }
  })
}

test('VTreeViewNode.ts', ({ mount }) => {
  let treeview

  beforeEach(() => {
    treeview = {
      register: jest.fn(),
      unregister: jest.fn()
    }
  })

  it('should return indeterminate icon', async () => {
    const wrapper = mount(VTreeviewNode, {
      provide: { treeview }
    })

    expect(wrapper.vm.computedIcon).toBe('$vuetify.icons.checkboxOff')

    wrapper.setData({ isIndeterminate: true })

    expect(wrapper.vm.computedIcon).toBe('$vuetify.icons.checkboxIndeterminate')
  })

  it('should use scoped slots', () => {
    const wrapper = mount(Mock, {
      provide: { treeview }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should generate a transition element', () => {
    const wrapper = mount(VTreeviewNode, {
      propsData: { transition: true },
      provide: { treeview }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
