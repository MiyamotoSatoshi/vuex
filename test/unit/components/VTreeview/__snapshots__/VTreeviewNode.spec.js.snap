// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VTreeViewNode.ts should generate a transition element 1`] = `

<div class="v-treeview-node v-treeview-node--leaf">
  <div class="v-treeview-node__root">
    <div class="v-treeview-node__content">
      <label class="v-treeview-node__label">
      </label>
    </div>
  </div>
</div>

`;

exports[`VTreeViewNode.ts should use scoped slots 1`] = `

<div class="v-treeview-node v-treeview-node--leaf">
  <div class="v-treeview-node__root">
    <div class="v-treeview-node__content">
      <div>
        foobar
      </div>
      <label class="v-treeview-node__label">
      </label>
      <div>
        foobar
      </div>
    </div>
  </div>
</div>

`;
